diff --git a/src/java/com/android/internal/telephony/cdma/CdmaDataProfileTracker.java b/src/java/com/android/internal/telephony/cdma/CdmaDataProfileTracker.java
index 3f1301a..9eeb53b 100644
--- a/src/java/com/android/internal/telephony/cdma/CdmaDataProfileTracker.java
+++ b/src/java/com/android/internal/telephony/cdma/CdmaDataProfileTracker.java
@@ -219,10 +219,10 @@ public final class CdmaDataProfileTracker extends Handler {
         }
 
         log("Loading data profiles for operator = " + operator);
-        String selection = "numeric = '" + operator + "'" + " and profile_type = 'nai'";
+        String selection = "numeric = '" + operator + "'";
         // query only enabled nai.
         // carrier_enabled : 1 means enabled nai, 0 disabled nai.
-        selection += " and carrier_enabled = 1";
+        //selection += " and carrier_enabled = 1";
         log("readNaiListFromDatabase: selection=" + selection);
 
         Cursor cursor = mPhone.getContext().getContentResolver().query(
diff --git a/src/java/com/android/internal/telephony/cdma/CdmaSubscriptionSourceManager.java b/src/java/com/android/internal/telephony/cdma/CdmaSubscriptionSourceManager.java
index 8cf5d29..60a22e4 100644
--- a/src/java/com/android/internal/telephony/cdma/CdmaSubscriptionSourceManager.java
+++ b/src/java/com/android/internal/telephony/cdma/CdmaSubscriptionSourceManager.java
@@ -46,7 +46,7 @@ public class CdmaSubscriptionSourceManager extends Handler {
     public static final int SUBSCRIPTION_SOURCE_UNKNOWN = -1;
     public static final int SUBSCRIPTION_FROM_RUIM      = 0; /* CDMA subscription from RUIM */
     public static final int SUBSCRIPTION_FROM_NV        = 1; /* CDMA subscription from NV */
-    public static final int PREFERRED_CDMA_SUBSCRIPTION = SUBSCRIPTION_FROM_NV;
+    public static final int PREFERRED_CDMA_SUBSCRIPTION = SUBSCRIPTION_FROM_RUIM;
 
     private static CdmaSubscriptionSourceManager sInstance;
     private static final Object sReferenceCountMonitor = new Object();
diff --git a/src/java/com/android/internal/telephony/uicc/RuimRecords.java b/src/java/com/android/internal/telephony/uicc/RuimRecords.java
index cfe7765..6d1345e 100755
--- a/src/java/com/android/internal/telephony/uicc/RuimRecords.java
+++ b/src/java/com/android/internal/telephony/uicc/RuimRecords.java
@@ -22,6 +22,8 @@ import static com.android.internal.telephony.TelephonyProperties.PROPERTY_ICC_OP
 
 import static com.android.internal.telephony.TelephonyProperties.PROPERTY_ICC_OPERATOR_ALPHA;
 import static com.android.internal.telephony.TelephonyProperties.PROPERTY_TEST_CSIM;
+import static com.android.internal.telephony.TelephonyProperties.PROPERTY_APN_SIM_OPERATOR_NUMERIC;
+
 
 import java.util.ArrayList;
 import java.util.Locale;
@@ -636,6 +638,7 @@ public final class RuimRecords extends IccRecords {
                 operator + "'");
         if (operator != null) {
             setSystemProperty(PROPERTY_ICC_OPERATOR_NUMERIC, operator);
+			setSystemProperty(PROPERTY_APN_SIM_OPERATOR_NUMERIC, operator);
         }
 
         if (mImsi != null) {
