diff --git a/liblight/lights.c b/liblight/lights.c
index 2c62744..b7224a4 100644
--- a/liblight/lights.c
+++ b/liblight/lights.c
@@ -15,8 +15,8 @@
  */
 
 
-// #define LOG_NDEBUG 0
-
+#define LOG_NDEBUG 0
+#define LOG_TAG "lights"
 #include <cutils/log.h>
 
 #include <stdint.h>
@@ -63,6 +63,12 @@ char const*const RED_BLINK_FILE
 char const*const LED_LOCK_UPDATE_FILE
         = "/sys/class/leds/red/device/lock";
 
+char const*const BUTTON_FILE = "/sys/class/leds/button-backlight/brightness";
+
+char const*const KEYBOARD_FILE = "/sys/class/leds/keyboard-backlight/brightness";
+
+char const*const BATTERY_FILE = "/sys/class/leds/red/brightness";
+
 /**
  * device methods
  */
@@ -95,6 +101,29 @@ write_int(char const* path, int value)
     }
 }
 
+static int write_str(char const *path, const char* value)
+{
+    int fd;
+    static int already_warned;
+
+    already_warned = 0;
+
+    ALOGV("write_str: path %s, value %s", path, value);
+    fd = open(path, O_RDWR);
+
+    if (fd >= 0) {
+        int amt = write(fd, value, strlen(value));
+        close(fd);
+        return amt == -1 ? -errno : 0;
+    } else {
+        if (already_warned == 0) {
+            ALOGE("write_str failed to open %s\n", path);
+            already_warned = 1;
+        }
+        return -errno;
+    }
+}
+
 static int
 is_lit(struct light_state_t const* state)
 {
@@ -229,6 +258,48 @@ set_light_attention(struct light_device_t* dev,
     return 0;
 }
 
+static int
+set_light_buttons(struct light_device_t* dev,
+        struct light_state_t const* state)
+{
+    int err = 0;
+    int on = is_lit(state);
+    ALOGD("===============> button.");
+    pthread_mutex_lock(&g_lock);
+    err = write_int(BUTTON_FILE, on?255:0);
+    pthread_mutex_unlock(&g_lock);
+
+    return err;
+
+}
+
+static int
+set_light_keyboard(struct light_device_t* dev,
+        struct light_state_t const* state)
+{
+    int err = 0;
+    int on = is_lit(state);
+    ALOGD("===============> keyboard.");
+    pthread_mutex_lock(&g_lock);
+    err = write_int(KEYBOARD_FILE, on?255:0);
+    pthread_mutex_unlock(&g_lock);
+
+    return err;
+}
+
+static int
+set_light_leds_battery(struct light_device_t* dev,
+        struct light_state_t const* state)
+{
+    int err = 0;
+    int on = is_lit(state);
+    ALOGD("===============> battery.");
+    pthread_mutex_lock(&g_lock);
+    err = write_str(BATTERY_FILE, on?"3":"0");
+    pthread_mutex_unlock(&g_lock);
+
+    return err;
+}
 
 /** Close the lights device */
 static int
@@ -253,13 +324,19 @@ static int open_lights(const struct hw_module_t* module, char const* name,
 {
     int (*set_light)(struct light_device_t* dev,
             struct light_state_t const* state);
-
+ALOGD("===============> %s.",name);
     if (0 == strcmp(LIGHT_ID_BACKLIGHT, name))
         set_light = set_light_backlight;
     else if (0 == strcmp(LIGHT_ID_NOTIFICATIONS, name))
         set_light = set_light_notifications;
     else if (0 == strcmp(LIGHT_ID_ATTENTION, name))
         set_light = set_light_attention;
+    else if (0 == strcmp(LIGHT_ID_BUTTONS, name))
+        set_light = set_light_buttons;
+    else if (0 == strcmp(LIGHT_ID_KEYBOARD, name))
+        set_light = set_light_keyboard;
+    else if (0 == strcmp(LIGHT_ID_BATTERY, name))
+        set_light = set_light_leds_battery;
     else
         return -EINVAL;
 
