diff --git a/quake/src/WinQuake/pr_exec.cpp b/quake/src/WinQuake/pr_exec.cpp
index 2af657a..39ce86c 100644
--- a/quake/src/WinQuake/pr_exec.cpp
+++ b/quake/src/WinQuake/pr_exec.cpp
@@ -577,7 +577,7 @@ while (1)
 #ifdef PARANOID
 		NUM_FOR_EDICT(ed);		// make sure it's in range
 #endif
-		a = (eval_t *)(ed->u.i + b->_int);
+		a = (eval_t *)((void *)(ed->u.i + b->_int));
 		c->_int = a->_int;
 		break;
 
@@ -586,7 +586,7 @@ while (1)
 #ifdef PARANOID
 		NUM_FOR_EDICT(ed);		// make sure it's in range
 #endif
-		a = (eval_t *)(ed->u.i + b->_int);
+		a = (eval_t *)((void *)(ed->u.i + b->_int));
 		c->vector[0] = a->vector[0];
 		c->vector[1] = a->vector[1];
 		c->vector[2] = a->vector[2];
